acvm/src/lib.rs

```diff
@@ -92,7 +97,26 @@ fn first_missing_assignment(
     })
 }
 
-pub trait Backend: SmartContract + ProofSystemCompiler + PartialWitnessGenerator + Default {}
+pub trait Backend:
+    SmartContract + ProofSystemCompiler + PartialWitnessGenerator + ReferenceString + Default
+{
+}
+
+#[async_trait]
+pub trait ReferenceString {
+    /// The Error type returned by failed function calls in the ReferenceString trait.
+    type Error: std::error::Error; // fully-qualified named because thiserror is `use`d at the top of the crate
+
+    /// Provides the reference string that is needed by other traits
+    async fn get_reference_string(&self, circuit: &Circuit) -> Result<Vec<u8>, Self::Error>;
```

Comment:
I would use common_reference_string instead of just reference_string, and I would call the function generate_common_reference_string(). 
The backend can cache it if he wants to, or download it, .. these are implementation details that are not relevant for the backend interface.

Reply:
Happy to rename it, but we're moving the caching into Nargo. Fetching or generating in some fashion is determine by the backend itself, but not filesystem access (helps with browser environment).

Reply:
Fetching or generating a CRS is done by a backend. Caching and filesystem access is done by Nargo.

Reply:
Renamed